### PR TITLE
Batch AUC

### PR DESCRIPTION
**Pull request by seiriosPlus**

### CODE REVIEW

**Review by jacquesqiao — DISMISSED** [review]
这个单测不太完整，后面还是完善一下

File: paddle/fluid/API.spec [hunk 1]
```diff
@@ -286,7 +286,7 @@ paddle.fluid.layers.iou_similarity ArgSpec(args=[], varargs='args', keywords='kw
 paddle.fluid.layers.box_coder ArgSpec(args=[], varargs='args', keywords='kwargs', defaults=None)
 paddle.fluid.layers.polygon_box_transform ArgSpec(args=[], varargs='args', keywords='kwargs', defaults=None)
 paddle.fluid.layers.accuracy ArgSpec(args=['input', 'label', 'k', 'correct', 'total'], varargs=None, keywords=None, defaults=(1, None, None))
-paddle.fluid.layers.auc ArgSpec(args=['input', 'label', 'curve', 'num_thresholds', 'topk'], varargs=None, keywords=None, defaults=('ROC', 4095, 1))
+paddle.fluid.layers.auc ArgSpec(args=['input', 'label', 'curve', 'num_thresholds', 'slide_steps', 'topk'], varargs=None, keywords=None, defaults=('ROC', 4095, 1, 1))
```
Comment (typhoonzero) [inline paddle/fluid/API.spec]: Put new arguments at the end to keep compability

Comment (seiriosPlus) [reply paddle/fluid/API.spec]: done

**Review by typhoonzero — APPROVED** [review]

File: paddle/fluid/operators/auc_op.cc [hunk 2]
```diff
@@ -62,25 +67,27 @@ class AucOpMaker : public framework::OpProtoAndCheckerMaker {
...
 
-    AddAttr<int>("num_thresholds",
-                 "The number of thresholds to use when discretizing the"
-                 " roc curve.")
+    AddAttr<int>(
+        "num_thresholds",
+        "The number of thresholds to use when discretizing the roc curve.")
         .SetDefault((2 << 12) - 1);
-
+    AddAttr<int>("slide_steps", "Use slide steps to calc batch auc.")
```
Comment (panyx0718) [inline paddle/fluid/operators/auc_op.cc:89]: This comment doesn't has any information...

**Review by panyx0718 — APPROVED** [review]
LG API

**Pull request merged**